### PR TITLE
Add support for S3 compatible endpoints

### LINKED ISSUE #10: Support Backblaze b2
Backblaze B2 is very similar to Amazon S3, but
• Operated by a less terrible company
• Cheaper

I use B2 for many things and would also like to use it with FileSafe.

### PR DESCRIPTION
**Pull request by pho**
This PR adds a provider selector on the S3 form
image
Since this is generic S3 now, I removed some AWS references in the code, but had to keep the integration name to avoid breaking current users' setup

I tested this against Scaleway Object Storage and its working nice.
This should close #10 too

I do not have an AWS nor B2 account, so I haven't been able to test those!

### CODE REVIEW

**Comment by mobitar** [comment]
Thanks for the PR 👌 I have this marked as unread in my inbox, so I'll get to it, but can't say precisely when.

**Comment by curiousleo** [comment]
I went looking for the repo precisely to find out if S3-compatible storage is supported (rather than just AWS), and this is the most recently merged PR -- nice.

It looks like there has not been a release since this was merged on 12 Feb, although I'm not sure. I just installed the FileSafe extension and in Standard Notes it says I'm on version 2.0.10 of FileSafe. The [releases page](https://github.com/standardnotes/filesafe-relay/releases) lists 1.2.12 as the most recent version, released on 13 Nov 2020.

So basically: this is a feature that I really want to use, and I'd love to see a version of FileSafe with this PR being released so I can start attaching documents to my notes :)

**Comment by JaspalSuri** [comment]
The change had to be reverted because it broke existing S3 integrations (for users using Amazon's S3 buckets). We would be interested in accepting (and testing) a potential fix for that issue. 🙂

That said, we are looking to replace FileSafe in the future by providing a first-party hosting solution for files.
